lychee-lib/src/checker/file.rs

```diff
@@ -134,8 +149,12 @@ impl FileChecker {
     ) -> Result<Cow<'a, Path>, ErrorKind> {
         let path = match path.metadata() {
             // for non-existing paths, attempt fallback extensions
+            // if fallback extensions don't help, try wikilinks
             Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
-                self.apply_fallback_extensions(path, uri).map(Cow::Owned)
+                match self.apply_fallback_extensions(path, uri).map(Cow::Owned) {
+                    Ok(val) => Ok(val),
+                    Err(_) => self.apply_wikilink_check(path, uri).map(Cow::Owned),
```

Comment:
The `Ok(val) => Ok(val)` looked a bit noisy. What do you think about this?

```suggestion
            Err(e) if e.kind() == std::io::ErrorKind::NotFound => {
                self.apply_fallback_extensions(path, uri)
                    .or_else(|_| self.apply_wikilink_check(path, uri))
                    .map(Cow::Owned)
            }
```